src/Illuminate/Cache/Repository.php

```diff
@@ -624,7 +624,9 @@ protected function getSeconds($ttl)
         $duration = $this->parseDateInterval($ttl);
 
         if ($duration instanceof DateTimeInterface) {
-            $duration = Carbon::now()->diffInSeconds($duration, false);
+            $duration = (int) ceil(
+                Carbon::now()->diffInMilliseconds($duration, false) / 1000
+            );
```

Comment:
I've changed this from `diffInSeconds` to `diffInMilliseconds / 1000` because Carbon 2 would always return an `int` while carbon 3 returns a float. This just allows us to ensure we get the correct value on both versions.